src/baldr/graphreader.cc

```diff
@@ -487,9 +557,22 @@ GraphReader::GraphReader(const boost::property_tree::ptree& pt,
                                                         pt.get<bool>("tile_url_gz", false));
   }
 
-  // validate tile url
-  if (!tile_url_.empty() && tile_url_.find(GraphTile::kTilePathPattern) == std::string::npos)
-    throw std::runtime_error("Not found tilePath pattern in tile url");
+  const bool is_tar_url =
+      !tile_url_.empty() && tile_url_.find(GraphTile::kTilePathPattern) == std::string::npos;
+
+  // some sanity checks
+  if (tile_url_.empty() &&
+      (tile_dir_.empty() || (!tile_dir_.empty() && !std::filesystem::exists(tile_dir_))) &&
+      tile_extract_->tiles.empty())
+    throw std::runtime_error("Couldn't find usable tileset source, check your config.");
```

Comment:
I thought it might be a good idea to add some sanity check if there's even any tiles locally or a URL specified, but seems I need to remove that again. There's a few tests which use a fake config and not use any tiles. They could probably be rewritten, but meeh..